### PR TITLE
aws_vpc: Correct the ARN account id

### PR DESCRIPTION
**Pull request by shuheiktgw**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates https://github.com/hashicorp/terraform-provider-aws/issues/16978

Output from acceptance testing:

```
$ TF_ACC=1 go test ./aws -v -count 1 -parallel 3 -run=TestAccAWSVpc_ -timeout 120m
=== RUN   TestAccAWSVpc_coreMismatchedDiffs
=== PAUSE TestAccAWSVpc_coreMismatchedDiffs
=== RUN   TestAccAWSVpc_basic
=== PAUSE TestAccAWSVpc_basic
=== RUN   TestAccAWSVpc_disappears
=== PAUSE TestAccAWSVpc_disappears
=== RUN   TestAccAWSVpc_ignoreTags
=== PAUSE TestAccAWSVpc_ignoreTags
=== RUN   TestAccAWSVpc_AssignGeneratedIpv6CidrBlock
=== PAUSE TestAccAWSVpc_AssignGeneratedIpv6CidrBlock
=== RUN   TestAccAWSVpc_Tenancy
=== PAUSE TestAccAWSVpc_Tenancy
=== RUN   TestAccAWSVpc_tags
=== PAUSE TestAccAWSVpc_tags
=== RUN   TestAccAWSVpc_update
=== PAUSE TestAccAWSVpc_update
=== RUN   TestAccAWSVpc_bothDnsOptionsSet
=== PAUSE TestAccAWSVpc_bothDnsOptionsSet
=== RUN   TestAccAWSVpc_DisabledDnsSupport
=== PAUSE TestAccAWSVpc_DisabledDnsSupport
=== RUN   TestAccAWSVpc_classiclinkOptionSet
=== PAUSE TestAccAWSVpc_classiclinkOptionSet
=== RUN   TestAccAWSVpc_classiclinkDnsSupportOptionSet
=== PAUSE TestAccAWSVpc_classiclinkDnsSupportOptionSet
=== CONT  TestAccAWSVpc_coreMismatchedDiffs
=== CONT  TestAccAWSVpc_update
=== CONT  TestAccAWSVpc_AssignGeneratedIpv6CidrBlock
--- PASS: TestAccAWSVpc_coreMismatchedDiffs (36.47s)
=== CONT  TestAccAWSVpc_tags
--- PASS: TestAccAWSVpc_update (73.99s)
=== CONT  TestAccAWSVpc_Tenancy
--- PASS: TestAccAWSVpc_AssignGeneratedIpv6CidrBlock (99.30s)
=== CONT  TestAccAWSVpc_classiclinkOptionSet
--- PASS: TestAccAWSVpc_tags (96.75s)
=== CONT  TestAccAWSVpc_classiclinkDnsSupportOptionSet
--- PASS: TestAccAWSVpc_classiclinkOptionSet (42.29s)
=== CONT  TestAccAWSVpc_ignoreTags
--- PASS: TestAccAWSVpc_Tenancy (95.87s)
=== CONT  TestAccAWSVpc_DisabledDnsSupport
--- PASS: TestAccAWSVpc_classiclinkDnsSupportOptionSet (42.54s)
=== CONT  TestAccAWSVpc_bothDnsOptionsSet
--- PASS: TestAccAWSVpc_ignoreTags (68.20s)
=== CONT  TestAccAWSVpc_basic
--- PASS: TestAccAWSVpc_DisabledDnsSupport (51.79s)
=== CONT  TestAccAWSVpc_disappears
--- PASS: TestAccAWSVpc_bothDnsOptionsSet (52.72s)
--- PASS: TestAccAWSVpc_basic (40.35s)
--- PASS: TestAccAWSVpc_disappears (29.25s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	252.910s

$ TF_ACC=1 go test ./aws -v -count 1 -parallel 1 -run=TestAccDataSourceAwsVpc_ -timeout 120m
=== RUN   TestAccDataSourceAwsVpc_basic
=== PAUSE TestAccDataSourceAwsVpc_basic
=== RUN   TestAccDataSourceAwsVpc_ipv6Associated
=== PAUSE TestAccDataSourceAwsVpc_ipv6Associated
=== RUN   TestAccDataSourceAwsVpc_multipleCidr
=== PAUSE TestAccDataSourceAwsVpc_multipleCidr
=== CONT  TestAccDataSourceAwsVpc_basic
--- PASS: TestAccDataSourceAwsVpc_basic (35.46s)
=== CONT  TestAccDataSourceAwsVpc_multipleCidr
--- PASS: TestAccDataSourceAwsVpc_multipleCidr (58.33s)
=== CONT  TestAccDataSourceAwsVpc_ipv6Associated
--- PASS: TestAccDataSourceAwsVpc_ipv6Associated (34.29s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	130.136s
```
Fix the VPC ARN account id to use the owner id.

### CODE REVIEW

File: aws/data_source_aws_vpc.go [hunk 1]
```diff
@@ -194,7 +194,7 @@ func dataSourceAwsVpcRead(d *schema.ResourceData, meta interface{}) error {
 		Partition: meta.(*AWSClient).partition,
 		Service:   ec2.ServiceName,
 		Region:    meta.(*AWSClient).region,
-		AccountID: meta.(*AWSClient).accountid,
+		AccountID: *vpc.OwnerId,
```
Comment (ewbankkit) [inline aws/data_source_aws_vpc.go:197]: References #12992.
```suggestion
		AccountID: aws.StringValue(vpc.OwnerId),
```

File: aws/resource_aws_vpc.go [hunk 2]
```diff
@@ -262,7 +262,7 @@ func resourceAwsVpcRead(d *schema.ResourceData, meta interface{}) error {
 		Partition: meta.(*AWSClient).partition,
 		Service:   ec2.ServiceName,
 		Region:    meta.(*AWSClient).region,
-		AccountID: meta.(*AWSClient).accountid,
+		AccountID: *vpc.OwnerId,
```
Comment (ewbankkit) [inline aws/resource_aws_vpc.go:265]: References #12992.
```suggestion
		AccountID: aws.StringValue(vpc.OwnerId),
```

**Comment by shuheiktgw** [comment]
@ewbankkit Fixed it, thank you!

**Review by ewbankkit — APPROVED** [review]
LGTM.

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSVpc_\|TestAccDataSourceAwsVpc_' ACCTEST_PARALLELISM=2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSVpc_\|TestAccDataSourceAwsVpc_ -timeout 120m
=== RUN   TestAccAWSVpc_coreMismatchedDiffs
=== PAUSE TestAccAWSVpc_coreMismatchedDiffs
=== RUN   TestAccDataSourceAwsVpc_basic
=== PAUSE TestAccDataSourceAwsVpc_basic
=== RUN   TestAccDataSourceAwsVpc_ipv6Associated
=== PAUSE TestAccDataSourceAwsVpc_ipv6Associated
=== RUN   TestAccDataSourceAwsVpc_multipleCidr
=== PAUSE TestAccDataSourceAwsVpc_multipleCidr
=== RUN   TestAccAWSVpc_basic
=== PAUSE TestAccAWSVpc_basic
=== RUN   TestAccAWSVpc_disappears
=== PAUSE TestAccAWSVpc_disappears
=== RUN   TestAccAWSVpc_ignoreTags
=== PAUSE TestAccAWSVpc_ignoreTags
=== RUN   TestAccAWSVpc_AssignGeneratedIpv6CidrBlock
=== PAUSE TestAccAWSVpc_AssignGeneratedIpv6CidrBlock
=== RUN   TestAccAWSVpc_Tenancy
=== PAUSE TestAccAWSVpc_Tenancy
=== RUN   TestAccAWSVpc_tags
=== PAUSE TestAccAWSVpc_tags
=== RUN   TestAccAWSVpc_update
=== PAUSE TestAccAWSVpc_update
=== RUN   TestAccAWSVpc_bothDnsOptionsSet
=== PAUSE TestAccAWSVpc_bothDnsOptionsSet
=== RUN   TestAccAWSVpc_DisabledDnsSupport
=== PAUSE TestAccAWSVpc_DisabledDnsSupport
=== RUN   TestAccAWSVpc_classiclinkOptionSet
=== PAUSE TestAccAWSVpc_classiclinkOptionSet
=== RUN   TestAccAWSVpc_classiclinkDnsSupportOptionSet
=== PAUSE TestAccAWSVpc_classiclinkDnsSupportOptionSet
=== CONT  TestAccAWSVpc_coreMismatchedDiffs
=== CONT  TestAccAWSVpc_Tenancy
--- PASS: TestAccAWSVpc_coreMismatchedDiffs (19.70s)
=== CONT  TestAccAWSVpc_classiclinkDnsSupportOptionSet
--- PASS: TestAccAWSVpc_classiclinkDnsSupportOptionSet (23.62s)
=== CONT  TestAccAWSVpc_classiclinkOptionSet
--- PASS: TestAccAWSVpc_Tenancy (64.43s)
=== CONT  TestAccAWSVpc_DisabledDnsSupport
--- PASS: TestAccAWSVpc_classiclinkOptionSet (22.79s)
=== CONT  TestAccAWSVpc_bothDnsOptionsSet
--- PASS: TestAccAWSVpc_DisabledDnsSupport (33.28s)
=== CONT  TestAccAWSVpc_update
--- PASS: TestAccAWSVpc_bothDnsOptionsSet (33.39s)
=== CONT  TestAccAWSVpc_tags
--- PASS: TestAccAWSVpc_update (49.85s)
=== CONT  TestAccAWSVpc_basic
--- PASS: TestAccAWSVpc_tags (60.72s)
=== CONT  TestAccAWSVpc_AssignGeneratedIpv6CidrBlock
--- PASS: TestAccAWSVpc_basic (22.53s)
=== CONT  TestAccAWSVpc_ignoreTags
--- PASS: TestAccAWSVpc_ignoreTags (41.64s)
=== CONT  TestAccAWSVpc_disappears
--- PASS: TestAccAWSVpc_AssignGeneratedIpv6CidrBlock (62.87s)
=== CONT  TestAccDataSourceAwsVpc_ipv6Associated
--- PASS: TestAccAWSVpc_disappears (13.70s)
=== CONT  TestAccDataSourceAwsVpc_multipleCidr
--- PASS: TestAccDataSourceAwsVpc_ipv6Associated (21.53s)
=== CONT  TestAccDataSourceAwsVpc_basic
--- PASS: TestAccDataSourceAwsVpc_basic (20.66s)
--- PASS: TestAccDataSourceAwsVpc_multipleCidr (44.28s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	269.795s
```

**Comment by ewbankkit** [comment]
@shuheiktgw Thanks for the contribution.
Could you please also update [this documentation](https://registry.terraform.io/providers/hashicorp/aws/latest/docs#skip_requesting_account_id) in `website/docs/index.html.markdown` to remove the affected resources and data sources.

**Comment by ewbankkit** [comment]
@shuheiktgw Minor conflict to resolve now that related PRs have been merged. Thanks.

**Comment by shuheiktgw** [comment]
@ewbankkit Thank you! Resolved the conflict 👍

**Pull request merged**